Shared/Realm.Shared/Realm.cs

```diff
@@ -985,5 +973,80 @@ public void Close()
         }
 
         #endregion
+
+        private class RealmState
+        {
+            #region static
+
+            #if __IOS__
+            [MonoPInvokeCallback(typeof(NativeCommon.NotifyRealmCallback))]
+            #endif
+            public static void NotifyRealmChanged(IntPtr stateHandle)
+            {
+                var gch = GCHandle.FromIntPtr(stateHandle);
+                ((RealmState)gch.Target).NotifyChanged(EventArgs.Empty);
```

Comment:
I don't think so - we keep the `GCHandle` to the state (line 1044) that is only freed after `realm.SharedRealmHandle.CloseRealm` is called (line 1075). I wouldn't expect that notifications would be delivered after calling `CloseRealm`. @fealebenpae you are more versed in OS inner workings, what's your take here?